Plugins/NativeEngine/Source/ShaderCompilerTraversers.cpp

```diff
@@ -237,9 +294,17 @@ namespace Babylon::ShaderCompilerTraversers
             {
             }
 
+            /// <summary>
+            /// Because no accumulation or cross-correlation is necessary for this change
+            /// (i.e. each operation acts only on a single symbol), we do all the work inside
+            /// the traverser itself.
+            /// </summary>
+            /// <param name="symbol"></param>
             void visitSymbol(TIntermSymbol* symbol) override
             {
                 auto& type = symbol->getType();
+
+                // We only care about uniforms that are neither samplers nor matrices.
```

Comment:
bgfx only supports 4x4 matrices. If we have some 3x3 rotation matrices, it has to be replaced as well. the c++ code already does the NxM to 4x4 matrix : https://github.com/BabylonJS/BabylonNative/blob/85e0731e2eab931f2e500996d75d3fa7a0d36bce/Plugins/NativeEngine/Source/NativeEngine.cpp#L1177

Reply:
Created #401 to track the work of proving and solving this issue.